pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/SymbolEquality.java

```diff
@@ -176,6 +177,25 @@ public boolean equals(JFormalParamSymbol f1, Object o) {
         }
     };
 
+
+    public static final EqAndHash<JRecordComponentSymbol> RECORD_COMPONENT = new EqAndHash<JRecordComponentSymbol>() {
+        @Override
+        public int hash(JRecordComponentSymbol t1) {
+            return 31 * t1.getEnclosingClass().hashCode() + t1.getSimpleName().hashCode();
+        }
+
+        @Override
+        public boolean equals(JRecordComponentSymbol f1, Object o) {
+            if (!(o instanceof JRecordComponentSymbol)) {
+                return false;
+            }
+            JRecordComponentSymbol f2 = (JRecordComponentSymbol) o;
+            return f1.nameEquals(f2.getSimpleName())
```

Comment:
```suggestion
            return Objects.equals(f1.getSimpleName(), f2.getSimpleName())
```

Reply:
Reason for this change: when reading this line, it is not clear that "nameEquals" actually means "simpleNameEquals" - better be explicit so that it is immediately clear that we compare the same attributes.

I'll change it for the other equal impls in this class as well. We used a mixture of nameEquals and Objects.equals already - so I'm changing it now for all to be Objects.equals.